pkg/apis/internal.admin.acorn.io/v1/computeclassresources.go

```diff
@@ -0,0 +1,150 @@
+package v1
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/acorn-io/baaah/pkg/typed"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// AllComputeClasses is a constant that can be used to define a ComputeResources struct that will apply to all
+// ComputeClasses. This should only be used when defining a ComputeClassResources struct that is meant to be used
+// as a limit and not a usage. The Fits method will work as expected when using this constant but Add and Remove
+// do not interact with it.
+const AllComputeClasses = "*"
+
+type ComputeResources struct {
+	Memory resource.Quantity `json:"memory,omitempty"`
+	CPU    resource.Quantity `json:"cpu,omitempty"`
+}
+
+func (current *ComputeResources) Equals(incoming ComputeResources) bool {
+	return current.Memory.Cmp(incoming.Memory) == 0 && current.CPU.Cmp(incoming.CPU) == 0
+}
+
+func (current *ComputeResources) ToString() string {
+	var resourceStrings []string
+
+	for _, r := range []struct {
+		resource string
+		value    resource.Quantity
+	}{
+		{"Memory", current.Memory},
+		{"CPU", current.CPU},
+	} {
+		switch {
+		case r.value.CmpInt64(0) > 0:
+			resourceStrings = append(resourceStrings, fmt.Sprintf("%s: %s", r.resource, r.value.String()))
+		case r.value.Equal(comparableUnlimitedQuantity):
+			resourceStrings = append(resourceStrings, fmt.Sprintf("%s: unlimited", r.resource))
+		}
+	}
+
+	return strings.Join(resourceStrings, ", ")
+}
+
+type ComputeClassResources map[string]ComputeResources
+
+// Add will add the ComputeClassResources of another ComputeClassResources struct into the current one.
+func (current ComputeClassResources) Add(incoming ComputeClassResources) {
```

Comment:
nit: if we define these methods on `*ComputeClassResources`, then the structure of the underlying map isn't copied every time they are called.

Reply:
spoke with tyler offline, but couldn't find supporting docs. what i did find was contrary to my statement; that copying a map is a effectively copying a single pointer instead of the map structure